quic: flaky TestResourceManagerAcceptDenied

```
 === RUN   TestResourceManagerAcceptDenied
      conn_test.go:49: using a RSA key: QmSVvBBCEK33xbuboVwD6mdDjEhDWUMr8UMCSerQkWykx5
      conn_test.go:49: using a Secp256k1 key: 16Uiu2HAmDkQQnsu9iuDwA4DnsAQ9pSffMgjJdCMiAMBxxwtAUzZ7
      transport.go:220: Unexpected call to *network.MockConnManagementScope.Done([]) at /home/runner/work/go-libp2p/go-libp2p/p2p/transport/quic/transport.go:220 because: there are no expected calls of the method "Done" for that receiver
  --- FAIL: TestResourceManagerAcceptDenied (1.71s)

--- a/p2p/transport/quic/conn_test.go
+++ b/p2p/transport/quic/conn_test.go
@@ -197,11 +197,12 @@ func TestResourceManagerAcceptDenied(t *testing.T) {
 	clientConnScope := mocknetwork.NewMockConnManagementScope(ctrl)
 	clientRcmgr.EXPECT().OpenConnection(network.DirOutbound, false).Return(clientConnScope, nil)
 	clientConnScope.EXPECT().SetPeer(serverID)
+	// In rare instances, the connection gating error will already occur on Dial.
+	// In that case, Done is called on the connection scope.
+	clientConnScope.EXPECT().Done().MaxTimes(1)
 	conn, err := clientTransport.Dial(context.Background(), ln.Multiaddr(), serverID)
 	// In rare instances, the connection gating error will already occur on Dial.
-	if err != nil {
-		clientConnScope.EXPECT().Done()
-	} else {
+	if err == nil {
 		_, err = conn.AcceptStream()
 		require.Error(t, err)
 	}

